File: src/main/java/org/example/chessplatformbe/persistence/CommentRepository.java
package org.example.chessplatformbe.persistence;

import org.example.chessplatformbe.domain.Article;
import org.example.chessplatformbe.domain.Comment;
import org.example.chessplatformbe.domain.User;

import java.util.List;
import java.util.Optional;

public interface CommentRepository {
    Optional<Comment> findById(Integer id);

    Comment save(Comment comment);

    void delete(Comment user);

    List<Comment> findByArticle_Id(Integer articleId);
}

File: src/persistence/comment_repository.rs
use crate::domain::comment::Comment;

pub trait CommentRepository {
    fn find_by_id(&self, id: u32) -> Option<Comment>;
    fn delete(&mut self, comment: Comment);
    fn find_by_article_id(&self, article_id: u32) -> Vec<Comment>;
    fn find_by_username(&self, username: &str) -> Vec<Comment>;
}
